students/junghoon/user/views.py

```diff
@@ -1,3 +1,41 @@
-from django.shortcuts import render
+import json, re
 
-# Create your views here.
+from django.http import JsonResponse
+from django.views import View
+
+from.models import User
+
+class SignupView(View):
+    def post(self, request):
+        try:
+            data                  = json.loads(request.body)
+            name                  = data['name']
+            password              = data['password']
+            email                 = data['email']
+            contact               = data['contact']
+            email_nomalization    = re.compile(
+                '^[a-zA-Z0-9-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-_]+\.*[a-zA-Z0-9-_]+$'
+            )
+            password_nomalization = re.compile(
+                '^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$'
+            )
+
+            if not re.match(email_nomalization, email):
+                return JsonResponse({'message': "EMAIL_IS_NOT_VALID"}, status=400)
+
+            if not re.match(password_nomalization, password):
+                return JsonResponse({"message": "PASSWORD_IS_NOT_VALID"}, status=401)
```

Comment:
1. 정규표현식을 사용할 경우, 자주 사용되는 경우가 아니라면 `compile()`해서 사용하지 않고, `match()`를 사용하는 것을 공식 문서에서 추천하고 있습니다.
2. 정규 표현식 같은 경우 변수라기 보다는 상수입니다. python에서 상수는 대문자로 만듭니다.

```suggestion
        REGEX_EMAIL    = '^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$'
        REGEX_PASSWORD = '^(?=.*[A-Za-z])(?=.*\d)(?=.*[$@$!%*#?&])[A-Za-z\d$@$!%*#?&]{8,}$'

        if not re.match(REGEX_EMAIL, data['email'):
            return JsonResponse({'message':'EMAIL_IS_NOT_VALID'}, status=400)
         
        if not re.match(REGEX_PASSWORD, data['password'):
            return JsonResponse({'message':'PASSWORD_IS_NOT_VALID'}, status=400)
```

Reply:
re.match 할때 컴파일을 수행해서 해줄 필요가 없다고 이해 했습니다.
감사합니다.